web/index.html

```diff
@@ -376,32 +437,36 @@ <h3>Used in</h3>
             return queryParams;
         }
 
-        // Decode base64 value mod 1000
-        function getChunk(value) {
-            let base64 = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-='
-            let result = 0
-            for (let i = 0; i < value.length; i++) {
-                result = (result * 64 + base64.indexOf(value[i]))
-            }
-            return Math.floor(result / 100)
-        }
-
-        const DATA_PREFIX = 'data/';
+        const PATH_PREFIX = window.location.host == 'expitau.github.io' ? '/' : '';
+        const CHUNK_WORKER_PATH = `${PATH_PREFIX}loadChunkWorker.js`;
+        const DATA_PATH = `${PATH_PREFIX}data/`;
+        const HELPER_PATH = window.location.host == 'expitau.github.io' ? 'https://expitau.github.io/InfiniteCraftWiki/helper'
+            : 'http://127.0.0.1:5501/web/helper.html';
 
         // Vue app
         const app = new Vue({
             el: '#app',
             data: {
                 data: { index: {}, list: [], costs: {}, craftingCache: {}, data: {}, metadata: { recipeCount: 0, loadedChunks: [] } },
+                chunkLoadState: [],
+                craftingTreeData: null,
+                bookmarkletData: "",
                 currentElement: "",
+                elementLoaded: false,
```

Comment:
This is the global state used to tell if the currently selected element has been loaded.
This is used to display the loading data text before it has been loaded, and then switch the From and To lists to the tables when things have been loaded and are ready to be viewed. Using a single variable like this is a lot more convenient than trying to pull that data from the ChunkLoadStatus array, as parsing Promise states is not fully consistent across browsers

Reply:
Okay, there isn't a way to do this fully imperatively, because that's just not how javascript async works. But I have refactored things such that `this.data.metadata.loadedChunks[chunk]` now holds an array of `[promise, bool]` where the first element can be awaited, and the second element is just the loaded state of that chunk. These values are only ever updated with each other in the load chunk function, and I created another method isChunkLoaded that returns if the current chunk is loaded or not, which is now used for the conditional rendering of elements.